app/models/agents/imap_folder_agent.rb

```diff
@@ -319,7 +320,7 @@ def each_unread_mail
         interpolated['folders'].each { |folder|
           log "Selecting the folder: %s" % folder
 
-          imap.select(folder)
+          imap.select(Net::IMAP.encode_utf7(folder))
```

Comment:
IMAP is interesting, utf7 does it mean one can not have emojis in IMAP folders? 😄

Reply:
UTF-7 can encode any character that can be encoded in UTF-16, so you can use emoji in IMAP folder names. (IMAP4 uses a slightly modified version of UTF-7, though)